packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -539,6 +539,25 @@ class AbciAppDB:
 
     The parameters cleanup_history_depth and cleanup_history_depth_current can also be configured in skill.yaml so they are used automatically
     when the cleanup method is called from AbciApp.cleanup().
+
+    # Memory warning
+    -----------------------------------
+    The database is implemented in such a way to avoid indirect modification of its contents.
+    It copies all the mutable data structures\\*, which means that it consumes more memory than expected.
+    This is necessary because otherwise it would risk chance of modification from the behaviour side,
+    which is a safety concern.
+
+    The effect of this on the memory usage should not be a big concern, because:
+
+        1. The synchronized data of the agents are not intended to store large amount of data.
+         IPFS should be used in such cases, and only the hash should be synchronized in the db.
+        2. The data are automatically wiped after a predefined `cleanup_history` depth as described above.
```

Comment:
Should we make clear that this is for historical data only, and the latest period data will be kept.

Reply:
This is clarified above:

https://github.com/valory-xyz/open-autonomy/blob/c16259e7295ce48f1c536a4c23227c802efa2280/packages/valory/skills/abstract_round_abci/base.py#L532-L541